### ISSUE TITLE
ZipArchive.TestArchive(true) fails after using IO.Compression.ZipFile.CreateFromDirectory() on an Folder with several small textfiles

### ISSUE THREAD

**Issue by Kalkihe** [icsharpcode/SharpZipLib#699]
### Steps to reproduce
1. Use the files in the attached zip to create an zip file using ZipFile.CreateFromDirectory() from the namespace System.Io.Compression.
2. Create a new ZipArchive-Object and test it using TestArchive.
`ZipFile zipFile = new ZipFile(pathToZipFile);
bool success = zipFile.TestArchive(true);
zipFile.Close();`
3. Check the return value. 

### Expected behavior
The return value should be true because it is a proper archive file.

### Actual behavior
The return value is always false. 

It turns out that, if I use SharpZibLib to generate the archive (which is not an option for the problem I'm working on), TestArchive returns true. It also succeeds, if I only put an database file of the size of ~500kb into the archive. But if I then add File1.txt and Subfolder1\Subfile2.txt (from Testfiles.zip), it returns false again. 

### Version of SharpZipLib
v1.1.0.145

### Obtained from (only keep the relevant lines)
- Package installed using NuGet

[Testfiles.zip](https://github.com/icsharpcode/SharpZipLib/files/7808080/Testfiles.zip)

**Comment by piksel** [comment]
I uploaded `Testfiles.zip` to archivediag to see what is going on, the report should appear here when done:
https://archivediagstore.blob.core.windows.net/reports/65y5sBy3REKdlAnYjIvDYg.html

**Comment by Kalkihe** [comment]
Actually, Testfiles.zip is just a manually generated archive of the folder-structure I used to create and validate an archive.

But with this post, I attached the actual resulting zip that fails to be validated. 
[SaveToZipHelperTest.zip](https://github.com/icsharpcode/SharpZipLib/files/7813419/SaveToZipHelperTest.zip)

**Comment by piksel** [comment]
Hm, that file seems to pass the test as well...

https://pub.p1k.se/sharpziplib/archivediag/SaveToZipHelperTest.zip.html

Perhaps it's not fully written when you are testing it?

**Comment by Kalkihe** [comment]
It is. I debugged my test step by step and before performing TestArchive, the archive is completely written and I can open it. Furthermore, I ran the archive check of 7zip on it, which also passes without any errors.

I think TestArchive may have a problem with that kind or archive. Sadly, there's no way to access any list of errors that function detects, Furthermore, the same test works, when I place different, slightly bigger data into the folder structure for that archive.

**Comment by piksel** [comment]
You can get the errors by supplying a logging callback to the method:
```cs
new ZipFile(archiveStream).TestArchive(true, TestStrategy.FindAllErrors, (status, message) =>
{
    if (status.Operation == TestOperation.EntryData) return;
    var pad = "";
    if (status.Entry is { } entry)
    {
        PrettyConsole.WriteLine($"[{status.Operation}{pad}] #{entry.ZipFileIndex} {entry.Name}");
    }
    else
    {
        PrettyConsole.WriteLine($"[{status.Operation}{pad}]");
    }

    if (!string.IsNullOrEmpty(message))
    {
        PrettyConsole.WriteLine($"[{status.Operation}{pad}] {message}");
    }
});
```
(this is from the tests done in ArchiveDiag, which actually returns `true` for the file).

> Version of SharpZipLib
> v4.0.30319

That's not an official SharpZipLib version... maybe you are using a fork?

**Comment by Kalkihe** [comment]
Thanks for your hint. I took another look and realized, I took the wrong version. Actually, I'm using version 1.1.0.145, which obviously is quiet outdated. In that project, it isn't referenced as nuget paket as I first thought, so I'm going to fix that issue.

I tried it out in a new project using the current nuget package alongside the same data. This works pretty well, so updating the package should solve the issue. 

Thanks for your help.
